File: 1-10/1-Two-Sum/C++/Program.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> hashMap;
        for (int i = 0; i < nums.size(); i++)
        {
            // int diff target - nums[i]
            if (hashMap.count(target - nums[i]))
                return {hashMap[target - nums[i]], i};
            hashMap[nums[i]] = i;
        }
    }
};

int main(int argc, char *argv[])
{
    Solution s;
    int target = 9;
    vector<int> nums{ 2, 7, 11, 15};
    int size = s.twoSum(nums, target).size();
    cout << "[ ";
    for (int i = 0; i < size; i++)
    {
         cout << s.twoSum(nums, target).at(i) << " ";
    }
    cout << "]";
    return 0;
}

File: 1-10/1-Two-Sum/Python/program.py
from ast import List
class Solution:
    def twoSum(self, nums, target):
        hashMap = {}
        for i, num in enumerate(nums):
            diff = target - num
            if diff in hashMap:
                return [hashMap[diff], i]
            hashMap[num] = i

if __name__ == '__main__':
    s = Solution()
    print (s.twoSum([2, 7, 11, 15], 9))